File: 3499-maximize-active-section-with-trade-i/3499-maximize-active-section-with-trade-i.cpp
class Solution {
public:
    int maxActiveSectionsAfterTrade(string s) {
        int res = 0;
        int prev = 0;
        int cur = 0;
        int seqCount = 0;
        int maxLength = 0;
        for(int i = 0; i < s.length(); i++)
        {
            if(s[i] == '1')
            {
                res++;
            }
        }
        s = "1" + s + "1";
        for(int i = 0; i < s.length(); i++)
        {
            if(s[i] == '0')
            {
                seqCount++;
                while(s[i] == '0')
                {
                    cur++;
                    i++;
                }
                if(seqCount >= 2)
                {
                    maxLength = max(maxLength, cur + prev);
                }
                prev = cur;
                cur = 0;
            }
        }
        res += maxLength;
        return res;
    }
};

File: 3499-maximize-active-section-with-trade-i/3499-maximize-active-section-with-trade-i.py
class Solution:
    def maxActiveSectionsAfterTrade(self, s: str) -> int:
        res = s.count('1')
        prev = 0
        cur = 0
        seqCount = 0
        maxLength = 0
        i = 0
        while i < len(s):
            if s[i] == '0':
                seqCount += 1
                while i < len(s) and s[i] == '0':
                    cur += 1
                    i += 1
                if seqCount >= 2:
                    maxLength = max(maxLength, cur + prev)
                prev = cur 
                cur = 0
            i += 1
        if seqCount >= 2:
            maxLength = max(maxLength, cur + prev)
        res += maxLength
        return res
